### PR TITLE
[21379] Design Issues on workflow screen

### PR DESCRIPTION
**Pull request by HDinger**
This applies a new design for the workflow-table. All icons are consistent (black) and the first header is now on the left side of the table. Thus the understandability shall be increased.

https://community.openproject.org/work_packages/21379

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hdinger-fix-21379-design-issues-workflow-screen.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hdinger-fix-21379-design-issues-workflow-screen.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by NobodysNightmare** [comment]
First of all: This looks rather nice! :+1:

Still there are some issues I found UI-wise (did not check the code yet). I think we could address all of those issues in a separate WP or fix them here. Just tell me your preference. I will create separate comments for the issues.

**Comment by NobodysNightmare** [comment]
### no column hovering

While I can hover over rows to highlight them, the usual hovering of columns seems to be broken:

broad-hover

When hovering inside the red rectangle, all columns are highlighted. Outside of this rectangle no columns are highlighted at all.

**Comment by NobodysNightmare** [comment]
### ambigious tooltips

When hovering over the tooltip for a single column or a single row, the tooltip shown (and displayed to impaired users) is "Check all/Uncheck all". This is the same as for the checkmark at the top right (the one that really checks and unchecks **all** other checkboxes).

I'd vote for more diversity:
- check/uncheck all
- check/uncheck (all in?) row
- check/uncheck (all in?) column

wrong-tooltip

**Comment by NobodysNightmare** [comment]
### no text for global checkmark

The checkmark in the top-right corner is quite lost. It has no text beside it and is far away from other UI elements.

global-checkmark

This time I have no clear idea what exactly should be changed, but I think probably an added text ("all"?) to the right would be helpful.

**Comment by NobodysNightmare** [comment]
### wrong size for headline in fieldset

**Reproduction**
1. (freshly reload the workflow page)
2. open one of the fieldsets below the main table
3. have a look at the now visible table (do not hover it with your mouse cursor!)

wrong-width

The layout of the headline ("new statuses allowed" and the global checkmark) is broken, their size is reset to the expected size as soon as you hover the table.

File: app/assets/stylesheets/content/_tables.sass [hunk 1]
```diff
@@ -58,6 +58,40 @@ table
     .hours-dec
       font-size: 0.9em
 
+  &[class*=transitions]
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_tables.sass]: I could imagine that this selector will match more often than anticipated. and you only want to match the three tables on the workflow page.

Can't you either specify all three classes explicitly or add another (common) class to all of them? The current selector will also match for `-smooth-transitions` etc.

**Comment by NobodysNightmare** [comment]
### IE 11

Just a reference screenshot from IE 11. Does not look too bad:

IE11

The only thing I noticed is that the headline and "check all" checkmark are left aligned in their spans, instead of right aligned.

**Comment by HDinger** [comment]
The remaining issues ('no column hovering' and 'wrong size for headline in fieldset') will be addressed in https://community.openproject.org/work_packages/21427

@NobodysNightmare can you please recheck this PR?

**Comment by NobodysNightmare** [comment]
@HDinger As discussed:
- check all/uncheck all is non-functional
- the links for check all/uncheck all should at least be colored

**Comment by NobodysNightmare** [comment]
Sorry :(

Not sure why I did not notice this earlier, but any chance that the filters on top look like this for you too:
bad

Instead of looking like this:
good

I did all the stuff to ensure that the fault is not on my side (rerunning `rake assets:webpack` etc), but for the `dev` branch and your first commit of this PR it looks fine. Starting at d612b38 ("Include review feedback") it is broken, regardless of my webpack attempts...

**Pull request merged**